boilerplate/update

```diff
@@ -196,6 +202,35 @@ for convention in $(scrubbed_conventions $CONFIG_FILE); do
   echo ""
 done
 
+# (Create and) edit .gitattributes to
```

Comment:
I'd really like to keep this as an optional config. Some consumers actually want to see the boilerplate diffs in their PRs.

Reply:
> I'd really like to keep this as an optional config. Some consumers actually want to see the boilerplate diffs in their PRs.

> You also have the potential of clobbering an existing .gitattributes.

We only clobber (with `sed`) and replace (by _appending_) the `GENERATED` section. It should be obvious that you shouldn't muck with that section, though I suppose I can add an extra comment in there. But as long as you don't, the only clobbering is caused by later directives overriding earlier ones. Which is 100% intentional because:

In order to instill confidence that the boilerplate changes in the PR don't deviate from what's in bp itself, we must also make sure that freeze-check itself hasn't been messed with. So we want to make sure it and its dependencies are revealed by default.

> Some consumers actually want to see the boilerplate diffs in their PRs.

Ah, so actually what we could do is leave the first directive (`boilerplate/** linguist-generated=true`) out of there, allowing consuming repos to decide whether they want it or not. If they do, the `GENERATED` section will still be applied after, to ensure certain deltas are revealed by default for the reasons described above. If they don't, the `GENERATED` section is a no-op.

Would that satisfy?

Reply:
I did these things.